internal/services/recoveryservices/site_recovery_vmware_replicated_vm_resource.go

```diff
@@ -0,0 +1,1204 @@
+		}
+
+		// Find first disk that is not fully replicated yet
+		if resp.Model.Properties.ProtectionState == nil {
+			return nil, "", fmt.Errorf("missing ProtectionState in response when making Read request on site recovery replicated vm Vmware Classic %s : %+v", id.String(), err)
+		}
+		return *resp.Model, *resp.Model.Properties.ProtectionState, nil
+	}
+}
+
+func fetchSiteRecoveryReplicatedVmVMWareContainerId(ctx context.Context, containerClient *replicationprotectioncontainers.ReplicationProtectionContainersClient, vaultId string) (string, error) {
+	vId, err := replicationprotectioncontainers.ParseVaultID(vaultId)
+	if err != nil {
+		return "", fmt.Errorf("parse %s: %+v", vaultId, err)
+	}
+
+	resp, err := containerClient.ListComplete(ctx, *vId)
+	if err != nil {
+		return "", err
+	}
+
+	if len(resp.Items) != 1 {
+		return "", fmt.Errorf("there should be only 1 protection container in Recovery Vault, get: %v", len(resp.Items))
+	}
+
+	parsedID, err := replicationprotectioncontainers.ParseReplicationProtectionContainerIDInsensitively(*resp.Items[0].Id)
+	if err != nil {
+		return "", fmt.Errorf("parse %s: %+v", *resp.Items[0].Id, err)
+	}
+
+	return parsedID.ID(), nil
+}
+
+func fetchRunAsAccountsIdBySite(ctx context.Context, runAsAccountClient *vmwarerunasaccounts.RunAsAccountsClient, siteId string, displayName string, applianceName string) (string, error) {
+	parsedSiteId, err := vmwarerunasaccounts.ParseVMwareSiteIDInsensitively(siteId)
+	if err != nil {
+		return "", fmt.Errorf("parse %s: %+v", siteId, err)
+	}
+
+	hackedClinet := azuresdkhacks.RunAsAccountsClient{Client: runAsAccountClient.Client}
+	// GET on Site is not working, tracked on https://github.com/Azure/azure-rest-api-specs/issues/24711
+	resp, err := hackedClinet.GetAllRunAsAccountsInSiteComplete(ctx, *parsedSiteId)
+	if err != nil {
+		return "", err
+	}
+
+	if len(resp.Items) == 0 {
+		return "", fmt.Errorf("retire run as account from %s, get 0 item", siteId)
+	}
+
+	for _, account := range resp.Items {
+		if account.Properties == nil {
+			continue
+		}
+		if account.Properties.DisplayName == nil {
+			continue
+		}
+		if account.Properties.ApplianceName == nil {
+			continue
+		}
+		if strings.EqualFold(*account.Properties.DisplayName, displayName) && strings.EqualFold(*account.Properties.ApplianceName, applianceName) {
+			return *account.Id, nil
+		}
+	}
+
+	return "", fmt.Errorf("retiring %q: run as account %s not found", siteId, displayName)
+}
+
+func fetchProcessServerIdByName(ctx context.Context, fabricClient *replicationfabrics.ReplicationFabricsClient, fabricId replicationfabrics.ReplicationFabricId, processServerName string) (string, error) {
+	resp, err := fabricClient.Get(ctx, fabricId, replicationfabrics.DefaultGetOperationOptions())
+	if err != nil {
+		return "", err
+	}
+
+	if resp.Model == nil {
+		return "", fmt.Errorf("retiring %q: Model is nil", fabricId)
```

Comment:
```suggestion
		return "", fmt.Errorf("retrieving %q: Model is nil", fabricId)
```

(Same as the other places)